vm/src/obj/objbyteinner.rs

```diff
@@ -387,6 +389,73 @@ impl PyByteInner {
 
         res
     }
+
+    pub fn count(
+        &self,
+        sub: Vec<u8>,
+        start: OptionalArg<PyObjectRef>,
+        end: OptionalArg<PyObjectRef>,
+        vm: &VirtualMachine,
+    ) -> Result<usize, PyObjectRef> {
+        let start = if let OptionalArg::Present(st) = start {
+            match_class!(st,
+            i @ PyInt => {Some(i.as_bigint().clone())},
+            _obj @ PyNone => None,
+            _=> {return Err(vm.new_type_error("slice indices must be integers or None or have an __index__ method".to_string()));}
+            )
+        } else {
+            None
+        };
+        let end = if let OptionalArg::Present(e) = end {
+            match_class!(e,
+            i @ PyInt => {Some(i.as_bigint().clone())},
+            _obj @ PyNone => None,
+            _=> {return Err(vm.new_type_error("slice indices must be integers or None or have an __index__ method".to_string()));}
+            )
+        } else {
+            None
+        };
+
+        let range = self.elements.get_slice_range(&start, &end);
+
+        if sub.is_empty() {
+            return Ok(self.len() + 1);
+        }
+
+        let mut total: usize = 0;
+        let mut i_start = range.start;
+        let i_end = range.end;
+
+        for i in self.elements.do_slice(range) {
+            if i_start + sub.len() <= i_end
+                && i == sub[0]
+                && &self.elements[i_start..(i_start + sub.len())] == sub.as_slice()
+            {
+                total += 1;
+            }
+            i_start += 1;
+        }
+        Ok(total)
+    }
+
+    pub fn join(&self, iter: PyIterable, vm: &VirtualMachine) -> PyResult {
+        let mut refs = vec![];
```

Comment:
Could you give a better name to this variable? Suggestions: `joined_bytes`, `result_bytes`?